content/en/blog/sign-verify-plugins-rules/index.md

```diff
@@ -0,0 +1,178 @@
+
+Upon request to install the artifact falcoctl is going to check the index file, resolve the name, identify the digest from the registry, pull the signature and validate it by using cosign as a library. Only once the signature is verified, falcoctl will allow the installation of the plugin with that specific digest. If the signature data does not match the installation will be halted with an error.
+
+You can try it yourself to make sure that the signature works!
+
+```
+sudo docker run -it --entrypoint sh falcosecurity/falcoctl:0.6.1
+# falcoctl artifact install cloudtrail:0.9 --rulesfiles-dir=/tmp --plugins-dir=/tmp
+```
+
+![](https://hackmd.io/_uploads/Byldjgnla.png)
+
+But there is more! If you have `cosign` installed you can independently verify the signature without using falcoctl or any index file:
+
+```
+$ cosign verify ghcr.io/falcosecurity/plugins/plugin/cloudtrail:0.9.0 --certificate-oidc-issuer=https://token.actions.githubusercontent.com --certificate-identity-regexp=https://github.com/falcosecurity/plugins/
+
+Verification for ghcr.io/falcosecurity/plugins/plugin/cloudtrail:0.9.0 --
+The following checks were performed on each of these signatures:
+  - The cosign claims were validated
+  - Existence of the claims in the transparency log was verified offline
+  - The code-signing certificate was verified using trusted certificate authority certificates
+
+[...]
+```
+
+This is one of the main adjvantages of relying on already existing and popular container artifacts signing technologies of the cloud native world. Authentication of the artifacts is decoupled from the producer, and consumers can always choose the method they prefer.
+
+### Security Analysis
+
+With the above scheme we made steps forward and implemented best practices towards better supply chain security in the Falco ecosystem. We would like to conclude our discussion with the main takeaways for a security practitioner.
+
+#### Safeguards against registry tampering, secure by default
+
+Making use of container registries to distribute artifacts is convenient, and allows for easy replication of the data across multiple regions or providers. But even if one of those registries is at risk of being compromised, like it [happened to AWS ECR in December 2022](https://www.bleepingcomputer.com/news/security/amazon-ecr-public-gallery-flaw-could-have-wiped-or-poisoned-any-image/), clients will be protected from potentially malicious applications. If **an artifact is replaced in the registry** without going through the appropriate approval and open source release pipeline, **clients will identify that the signature does not match and automatically refuse to install** any compromised package. This is only possible because cosign signature information is distributed in a file which is separate from the artifact itself. The feature is on by default and requires no extra action from users.
```

Comment:
```suggestion
Making use of container registries to distribute artifacts is convenient, and allows for easy replication of the data across multiple regions or providers. But even if one of those registries is at risk of being compromised, like it [happened to AWS ECR in December 2022](https://www.bleepingcomputer.com/news/security/amazon-ecr-public-gallery-flaw-could-have-wiped-or-poisoned-any-image/), clients will be protected from potentially malicious applications. If **an artifact is replaced in the registry** without going through the appropriate approval and open source release pipeline, **clients will identify that the signature does not match and automatically refuse to install** any compromised package. This is only possible because cosign signature information is distributed in a file that is separate from the artifact itself. The feature is on by default and requires no extra action from users.
```